File: mobile/android/core/designsystem/src/main/kotlin/com/slopssaloon/omen/core/designsystem/component/OmenStateSurface.kt
package com.slopssaloon.omen.core.designsystem.component

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.semantics.stateDescription
import com.slopssaloon.omen.core.designsystem.theme.OmenTheme

/** The six honest registry §3.1 data states. Never collapse one into another. */
enum class OmenStateSurfaceKind { Empty, Loading, Error, Disconnected, Stale, Mock }

/**
 * Token-backed, non-interactive state surface for content areas. Callers supply contextual copy
 * (for example, "Analyzing your matchup…"), never a generic dead-dashboard "Loading…" label.
 * [reducedMotion] renders loading as a static status instead of an animated spinner.
 */
@Composable
fun OmenStateSurface(
    kind: OmenStateSurfaceKind,
    title: String,
    message: String,
    modifier: Modifier = Modifier,
    reducedMotion: Boolean = false,
) {
    val cardVariant = when (kind) {
        OmenStateSurfaceKind.Empty -> OmenCardVariant.Empty
        OmenStateSurfaceKind.Error -> OmenCardVariant.Error
        else -> OmenCardVariant.Outlined
    }
    val tone = when (kind) {
        OmenStateSurfaceKind.Error -> OmenCardTone.Risk
        else -> OmenCardTone.Neutral
    }
    val stateLabel = kind.name.lowercase().replaceFirstChar { it.titlecase() }

    OmenCard(
        modifier = modifier
            .fillMaxWidth()
            .semantics { stateDescription = stateLabel },
        variant = cardVariant,
        tone = tone,
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(OmenTheme.spacing.step8)) {
            if (kind == OmenStateSurfaceKind.Loading && !reducedMotion) {
                CircularProgressIndicator(
                    color = OmenTheme.color.accent,
                    trackColor = OmenTheme.color.surface3,
                )
            }
            Text(title, style = OmenTheme.typography.h2.toTextStyle(), color = OmenTheme.color.textPrimary)
            Text(message, style = OmenTheme.typography.body.toTextStyle(), color = OmenTheme.color.textSecondary)
        }
    }
}

File: mobile/ios/OmenIOS/OmenIOS/DesignSystem/OmenStateSurface.swift
import SwiftUI

/// The six honest registry §3.1 content states. Never substitute one state for another.
enum OmenStateSurfaceKind { case empty, loading, error, disconnected, stale, mock }

/// Token-backed, non-interactive content state. Supply contextual loading copy rather than a
/// generic "Loading…" label. `reduceMotion` replaces the spinner with a static state.
struct OmenStateSurface: View {
    let kind: OmenStateSurfaceKind
    let title: String
    let message: String
    var reduceMotion = false

    private var cardVariant: OmenCardVariant {
        switch kind {
        case .empty: .empty
        case .error: .error
        default: .outlined
        }
    }

    private var stateLabel: String {
        switch kind {
        case .empty: "Empty"
        case .loading: "Loading"
        case .error: "Error"
        case .disconnected: "Disconnected"
        case .stale: "Stale"
        case .mock: "Mock"
        }
    }

    var body: some View {
        OmenCard(variant: cardVariant, tone: kind == .error ? .risk : .neutral) {
            VStack(alignment: .leading, spacing: OmenSpacing.step8) {
                if kind == .loading && !reduceMotion {
                    ProgressView().tint(OmenColor.accent)
                }
                Text(title).omenTextStyle(OmenTypography.h2).foregroundStyle(OmenColor.textPrimary)
                Text(message).omenTextStyle(OmenTypography.body).foregroundStyle(OmenColor.textSecondary)
            }
        }
        .accessibilityElement(children: .combine)
        .accessibilityValue(stateLabel)
    }
}
